lib/pbench/agent/tool_meister.py

```diff
@@ -1311,125 +1355,84 @@ def _send_directory(self, directory, uri, ctx):
         target_dir = directory.name
         parent_dir = directory.parent
         tar_file = parent_dir / f"{target_dir}.tar.xz"
-        o_file = parent_dir / f"{target_dir}.tar.out"
-        e_file = parent_dir / f"{target_dir}.tar.err"
+
         try:
-            # Invoke tar directly for efficiency.
-            with o_file.open("w") as ofp, e_file.open("w") as efp:
-                cp = subprocess.run(
-                    [
-                        self.tar_path,
-                        "--create",
-                        "--xz",
-                        "--force-local",
-                        f"--file={tar_file}",
-                        target_dir,
-                    ],
-                    cwd=parent_dir,
-                    stdin=None,
-                    stdout=ofp,
-                    stderr=efp,
-                )
+            if self._create_tar(directory, tar_file, retry=True).returncode != 0:
+                # Tar ball creation failed even after suppressing all the warnings,
+                # we will now proceed to create an empty tar ball.
+                # TODO: it'd be better to be able to skip the PUT entirely if the
+                #  tar fails and simply log a failure without TDS waiting forever
+                if self._create_tar(Path("/dev/null"), tar_file, retry=True) != 0:
+                    raise ToolMeisterError
```

Comment:
You can add message text to the `ToolMeisterError`, and I would recommend doing so.

I would also recommend adding a comment here which points out that, when we raise this exception, we're going to skip the `PUT`, below, and that's going to cause the TDS to hang....